src/Workspaces/Core/Portable/Shared/Extensions/SafeHandleLease.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    /// <summary>
+    /// Represents a lease of a <see cref="SafeHandle"/>.
+    /// </summary>
+    /// <seealso cref="SafeHandleExtensions.Lease"/>
+    internal readonly struct SafeHandleLease : IDisposable
+    {
+        private readonly SafeHandle _handle;
+
+        internal SafeHandleLease(SafeHandle handle)
+            => _handle = handle;
+
+        public void Dispose()
+            => _handle?.DangerousRelease();
```

Comment:
`?.` scares me here.  necessary?   can we NRT this and doc why null woudl be expected here?

Reply:
The containing type is a value type. The fields can always be null.

Reply:
it feels like it would be a bug for anyone to have a default-instance though, and if they called .Dispose i would prefer we die/assert rather than just silently ignore it.  

can we at least put an assert here?

Reply:
It's a common pattern for releasers to return a default instance if for some reason there will not be anything to release. I can't rule out the scenario with enough confidence to claim that it's an invalid use of the type.

Reply:
wfm.